### PR TITLE
fix: handled error state when cyclonedx dependencies are missing a depends on property

### PR DESCRIPTION
**Pull request by robh-snyk**
`guacone` is failing to parse CycloneDX documents where dependencies are missing a dependsOn property.

From the [CycloneDX specification component definition](https://github.com/CycloneDX/specification/blob/ccbf7b5781ef534cd62616e3c4221004c7c82a66/schema/bom-1.4.schema.json#L895-L920), this is not a required property, and should not raise an exception.

### CODE REVIEW

**Comment by lumjjb** [comment]
hi @robh-snyk , you may need to rebase on main since there's been some changes with the [testdata refactor](#261)

File: pkg/ingestor/parser/cyclonedx/parser_cyclonedx_test.go
Comment (pxp928) [inline pkg/ingestor/parser/cyclonedx/parser_cyclonedx_test.go:67]: Looks like you accidentally added  `CycloneDXDependenciesMissingDependsOn`. This will fix the error.

**Comment by pxp928** [comment]
> @lumjjb @pxp928 - I've updated the branch, however the integration tests are failing due to an OSV timeout - is this expected behaviour? Should I just re-run this?

Hey @robh-snyk I just kicked off the re-run.

**Comment by pxp928** [comment]
hmm strange that it's failing. Let me look into it.

**Comment by pxp928** [comment]
> > hmm strange that it's failing. Let me look into it.
> 
> 
> 
> Looks like the upstream OSV endpoint is randomly failing. A manual check got the following:
> 
> ```
> 
> curl -X POST -d '{"commit": "6879efc2c1596d11a6a6ad296f80063b558d5e0f"}' \
> 
>     "https://api.osv.dev/v1/query"
> 
> 
> 
> {"code":504,"message":"upstream request timeout"}
> 
> ```

Escalated the issue upstream.

**Review by lumjjb — APPROVED** [review]
LGTM! Thanks @robh-snyk ! Looks like it may be worth checking if some of these fields are nil as well before dereferencing them like `*cdxBom.Components`. FYI @nadgowdas

**Comment by lumjjb** [comment]
That would be a separate PR, we just need 2 approvals to get this merged, so waiting on 1 more maintainer review!!

**Review by pxp928 — APPROVED** [review]
LGTM

**Pull request merged**
